runtime/compiler/env/J9SharedCache.cpp

```diff
@@ -106,34 +106,37 @@ TR_J9SharedCache::TR_J9SharedCache(TR_J9VMBase *fe)
    _sharedCacheConfig = _javaVM->sharedClassConfig;
    _numDigitsForCacheOffsets = 8;
 
-   TR_ASSERT_FATAL(_sharedCacheConfig, "Must have _sharedCacheConfig");
-  
-   UDATA totalCacheSize = 0;
-   J9SharedClassCacheDescriptor *curCache = _sharedCacheConfig->cacheDescriptorList;
-   do
+#if defined(JITSERVER_SUPPORT)
```

Comment:
Can we keep the fatal assert when we are not in JITServer mode?
TR_ASSERT_FATAL(_sharedCacheConfig || _compInfo->getPersistentInfo()->getRemoteCompilationMode() == JITServer::SERVER, "Must have _sharedCacheConfig");